### ISSUE TITLE
[Accessibility] Cannot trace collaborators' cursor location

### ISSUE THREAD

**Issue by jooyoungseo** [iamhyc/Overleaf-Workshop#44]
Type: <b>Bug</b>

Thanks for creating a wonderful extension. This has a lot of potential for those who cannot use the original Overleaf UI. Blind people who use screen reader have some critical issues with the Overleaf's inaccessible web interface and its editor, and this kind of VSCode extension can offer an alternative and accessible workaround!

Although the extension is usable itself, I was wondering if you could consider some more accessible features implemented in `ms-vsliveshare.vsliveshare`. For example, currently, there is no easy way for screen reader users to identify or follow collaborator's cursor location.

It would be great if we had some features like teathering user's cursor with collaborator's cursor and jumping to a collaborator location.

This will contribute to improving the accessibility of collabrotive editing a lot!

Extension version: 0.5.1
VS Code version: Code - Insiders 1.85.0-insider (c6b48c364dde70c12bf977f1da50acda0fd801e3, 2023-11-30T05:35:40.559Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz (8 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.71GB (6.40GB free)|
|Process Argv|--crash-reporter-id b05b88e5-8894-4031-ae34-fa034ebddea9|
|Screen Reader|yes|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
py29gd2263:30784851
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
a9j8j154:30646983
showlangstatbar:30737417
fixshowwlkth:30771523
showindicator:30805243
pythongtdpath:30726887
i26e3531:30792625
welcomedialog:30812478
pythonnosmt12:30779711
pythonidxpt:30768918
pythonnoceb:30776497
asynctok:30898717
dsvsc013:30777762
dsvsc014:30777825
pythonmpswarning:30901777
dsvsc015:30821418
pythontestfixt:30866404
pythonregdiag2:30902439
pyreplss1:30879911
pythonmypyd1:30859725
pythoncet0:30859736
pythontbext0:30879054
accentitlementst:30870582
dsvsc016:30879898
dsvsc017:30880771
dsvsc018:30880772
aa_t_chat:30882232

```

</details>

**Comment by iamhyc** [comment]
Thanks for the suggestions.

I have not used the Live Share before. I will first do the survey to see how it works.


As for the screen reader being able to track the cursors' location, I am not sure if it is possible with the limitations of the vscode API.
Currently, these "cursors" are actually not real cursors, but just "zero-width rectangles" plotted at the character position.

**Comment by iamhyc** [comment]
Hi @jooyoungseo , I have read the doc you shared for the accessibility feature in vscode live share.
I can tell that it basically enhance the accessibility via playing sounds to notify various events.
Unfortunately, as vscode does not provide any public API for audio or accessibility now, I am not going to exhaust myself with the sound features.

Instead, I can add the cursor jump and tether functions to improve the experience for screen reader users.
I will first add the "jump to collaborators" function via command palette in the next release `v0.5.3`, which is easy to implement.
The tethering thing requires a smooth and consistent update of cursor position in relation to editing, which may require extras works, and I plan to do it later.
